core/unix/signal.c

```diff
@@ -3064,6 +3068,26 @@ fixup_rtframe_pointers(dcontext_t *dcontext, int sig, sigframe_rt_t *f_old,
                    sizeof(xstate_new->xstate_hdr));
             memcpy(&xstate_new->ymmh, &xstate_old->ymmh, sizeof(xstate_new->ymmh));
         }
+#    ifdef X64
+        if (ZMM_ENABLED()) {
+            kernel_xstate_t *xstate_new = (kernel_xstate_t *)tgt;
+            kernel_xstate_t *xstate_old =
+                (kernel_xstate_t *)f_old->uc.uc_mcontext.fpstate;
+            memcpy((byte *)xstate_new + proc_xstate_area_zmm_hi256_offs(),
+                   (byte *)xstate_old + proc_xstate_area_zmm_hi256_offs(),
+                   proc_num_simd_sse_avx_registers() * ZMMH_REG_SIZE);
+            memcpy((byte *)xstate_new + proc_xstate_area_hi16_zmm_offs(),
+                   (byte *)xstate_old + proc_xstate_area_hi16_zmm_offs(),
+                   proc_num_simd_sse_avx_registers() * ZMM_REG_SIZE);
```

Comment:
So "zmm_hi256" is the upper part of the zmm0-15 registers.  And "hi16_zmm" is the entire part of the new registers zmm16-31.  So you're assuming the new registers have the same count as the old?  I would think it's better to instead take the number of avx512 simd registers and subtract off the legacy ones.

Reply:
Done. That's actually what's being done in other parts of the patch. Thanks.